custom/live-streaming/components/LiveStreamingModal.js

```diff
@@ -18,35 +18,53 @@ const LAYOUTS = [
   { label: 'Active participant', value: 'active-participant' },
 ];
 
+const ParticipantOption = ({ sessionId }) => {
+  const participant = useParticipant(sessionId);
+
+  return (
+    <option value={participant.session_id} key={sessionId}>
+      {participant.user_name}
+    </option>
+  );
+};
+
 export const LiveStreamingModal = () => {
-  const { callObject } = useCallState();
-  const { allParticipants } = useParticipants();
+  const { participantIds } = useParticipants();
   const { currentModals, closeModal } = useUIState();
-  const { isStreaming, streamError } = useLiveStreaming();
+  const { isLiveStreaming, errorMsg, startLiveStreaming, stopLiveStreaming } =
+    useLiveStreaming();
+
   const [pending, setPending] = useState(false);
   const [rtmpUrl, setRtmpUrl] = useState('');
-  const [layout, setLayout] = useState(0);
+  const [layoutType, setLayoutType] = useState('default');
   const [maxCams, setMaxCams] = useState(9);
-  const [participant, setParticipant] = useState(0);
+  const [participantId, setParticipantId] = useState(0);
 
   useEffect(() => {
     // Reset pending state whenever stream state changes
     setPending(false);
-  }, [isStreaming]);
+  }, [isLiveStreaming]);
 
   function startLiveStream() {
     setPending(true);
 
-    const opts =
-      layout === 'single-participant'
-        ? { session_id: participant.id }
-        : { max_cam_streams: maxCams };
-    callObject.startLiveStreaming({ rtmpUrl, preset: layout, ...opts });
+    const config = {
+      rtmpUrl,
+      layout: {
+        preset: layoutType,
+      },
+    };
+
+    if (layoutType === 'single-participant')
+      config.layout.session_id = participantId;
```

Comment:
Nit: I get that sometimes single-line conditionals without braces read smoother, but imo _multiline_ conditionals usually read easier and more consistently with braces around them.